Underscore/USArrayWrapper.m

```diff
@@ -243,6 +243,19 @@ - (USArrayWrapper *)shuffle
     };
 }
 
+- (USArrayWrapper *)zip
+{
+    int length = (int)[self.first count];
+    NSMutableArray *finalMutableArray = [NSMutableArray arrayWithCapacity:length * self.array.count];
+    for (NSUInteger index = 0; index < length; index++) {
+        USArrayWrapper *wrapper = self.map(^id (id obj) {
```

Comment:
Why not: 

```ObjC
[finalMutableArray addObject: self.map(^id (id obj) {
    return obj[index];
}).unwrap];
```

I don't think having it pulled apart really adds any clarity in this case. In fact I found it a bit harder to follow with the intermediate wrapper variable being used.